tree/node.go

```diff
@@ -0,0 +1,29 @@
+package tree
+
+import "golang.org/x/crypto/sha3"
+
+type Node interface {
+	GetChild(key []Unit) Node
+	Insert(key []Unit, value []byte)
+	Delete(key []Unit) bool
+	SetChild(node Node)
+	SetParent(b Node)
+	Value() []byte
+	Hash()
+	GetHash() []byte
+	Key() []Unit
+	Print()
+}
+
+func Hash(bs ...[]byte) []byte {
```

Comment:
I think you can use our `hashing.ByteArraysToHash256Array` here instead

Reply:
Yeah, that's something I'd like to discuss further. Would like to have a few people review this hashing.

I'll change it to use the `hashing.ByteArraysToHash256Array` and I'll circle back to triple check :)